docs/rfcs/9999-vpn-firefox-integration.md

```diff
@@ -0,0 +1,188 @@
+    - [@lesleyjanenorton](https://github.com/lesleyjanenorton)
+    - [@oskirby](https://github.com/oskirby)
+- RFC PR: -
+- Implementation GitHub issue: -
+- [Design @Figma](https://www.figma.com/file/s13B7zs27cadZXUyvxobgW/Better-Together-Explorations?type=design&node-id=259-4830&mode=design&t=4BmtgXMzjnubd9gc-0)
+
+-----
+# Firefox + VPN Better together
+## Problem Statement
+
+
+![](https://github.com/mozilla-mobile/mozilla-vpn-client/assets/9611612/aa8331cf-f0cb-489b-a7e2-293bce422551) 
+
+VPNs currently allow you to select a City through which you want your traffic to emerge to the Internet and users use this to express a variety of geolocation preferences. These controls are not flexible enough - users can only select a single location through which all traffic should go through. 
+
+Therefore we want to propose a new Web-Extension that extends on top of the VPN+MAC API, giving users the options to:
+- Enable the VPN only on Firefox 
+ - Disable the VPN only on Firefox 
+ - Exclude a Firefox tab from VPN protection  
+ - Specify an Exit Location a Page.
+
+We already have integrated into Firefox with [Multi Account Containers](https://github.com/mozilla/multi-account-containers) which allows users to specify an exit location for a container. 
+The following parts expect knowledge on how the MAC Extension works, for a refresher scroll to appendix :) 
+
+----
+
+## Preconditions
+As we currently not plan to be a bundled feature into Firefox the first MVP, Users need to have the VPN Client installed, 
+be logged in with a subscription. During an ✨ Onboarding Flow ✨ (currently unspecified) users installed 
+the VPN-Integration Extension into their Firefox Profile. 
+
+## Delivery
+As the Extension will be Mozilla signed, asking Firefox to open a URL pointing to our XPI will prompt the install, without warnings. 
+The onboarding will be refined based on foxfooding feedback. 
+
+
+## VPN API 
+
+
+### Networking
+Using the [proxy api](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/proxy) a network extension can decide which proxy each network request should use. So from a network request perspective, there are only 4 possible cases to implement:
+
+| StateNr               | Device VPN is On | Device VPN is OFF |
+|-----------------|------------------|-------------------|
+| Page VPN is On  | 1                | 2                 |
+| Page VPN is Off | 3                | 4                 |
+
+In the following sections, we will explain what changes we need inside the client to support each of the cases and where the extension must proxy traffic to. 
+
+
+#### State 1: Device VPN is On & Page VPN is On
+In this state the extension can simply set the proxy for those network requests to `null` to use the user's active server.  
+If a custom exit location is set it must set it to the `socks5` proxy of that endpoint. No changes are needed on the client as this is how MAC uses the current API.

```

Comment:
> If a custom exit location is set it must set it to the `socks5` proxy of that endpoint.

Could some more information be added on how a custom exit location will be handled by setting it to the 'socks5' proxy, for the benefit of the reader?